errors/errors.go

```diff
@@ -0,0 +1,276 @@
+package errors
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// TritonError represents an error code and message along with
+// the status code of the HTTP request which resulted in the error
+// message. Error codes used by the Triton API are listed at
+// https://apidocs.joyent.com/cloudapi/#cloudapi-http-responses
+type TritonError struct {
+	StatusCode int
+	Code       string `json:"code"`
+	Message    string `json:"message"`
+}
+
+// Error implements interface Error on the MantaError type.
+func (e MantaError) Error() string {
+	return fmt.Sprintf("%s: %s", e.Code, e.Message)
+}
+
+// MantaError represents an error code and message along with
+// the status code of the HTTP request which resulted in the error
+// message. Error codes used by the Manta API are listed at
+// https://apidocs.joyent.com/manta/api.html#errors
+type MantaError struct {
```

Comment:
Same comment but for `MantaError`.  ?

.. actually, evolving thought process here.  Is this thrust correct?  I understand wanting to centralize the client errors into a single package, that makes sense.  But in terms of responses that come back from the API, I'm not sure this is right.  A Compute error may share a message with a Manta error, but the two aren't interchangeable types.  Wouldn't we want to do something like:

- Add `compute/errors.go`
- Add `compute.Error` as a type
- Add `manta.Error` as a type

Different packages are going to have differing levels of detail they may want to expose, but in all cases, none of the error types are fungable.  We may want to have it so that the client exposes an error type that satisfies a `TritonError` `interface` that can be type-asserted into `compute.Error` or `manta.Error`, but this feels like we're trying to create a single straight-jacket for all errors to fit into.

And to be clear, the current error handling has been sufficient, but isn't ideal, so I'm glad we're doing this.  But I'm not sure we're doing the right thing quite yet.

I just went to look at what the Google API client does (they seem like an informed bunch w.r.t. error interface design... hrm...):

https://github.com/google/google-api-go-client/blob/a11cf47576f3bc79d7e517bf21887f4c390b1a69/googleapi/googleapi.go#L61

Yeah, that seems passable.  The have a slice of errors from the API.  All APIs return a consistent error type.  Each service endpoint, however, doesn't have its own error type:

https://github.com/google/google-api-go-client/blob/3a1d936b7575b82197a1fea0632218dd07b1e65c/pubsub/v1/pubsub-gen.go#L1477-L1495

I wonder if that's because this code base is transpiled from some other service definition.  Let's take this offline.  I'm not entirely convinced of my own arguments and am open to debate.

My instinct says each discrete component of Triton should have its own error type with extra details and that each component error type should implement a standard interface so that clients don't need to be bothered with the details unless they want the supplemental details that `compute` may provide, for example.